### ISSUE TITLE
Renaming a database keeps the old one around

### ISSUE THREAD

**Issue by maknz** [sequelpro/sequelpro#2334]
As of the most recent version, when you rename a database, an empty database with the old name is retained. The behaviour previously was that when you renamed a database, it had the effect of just changing the name.

**Comment by dmoagx** [comment]
This is intentional because Sequel Pro right now can't correctly determine if all objects were moved and deleting the db with objects left would mean data loss.

**Comment by firewaller** [comment]
Can you detect if the old database is empty (preserving databases with data) after rename and then delete?

**Comment by stefanbauer** [comment]
This Bug is still there. Nightly 4667

**Comment by philcook** [comment]
Understand that this relies on #1491 but this is quite annoying having to delete the database after a rename.

Can I suggest whilst awaiting for #1491 there is an option added for "rename and delete" underneath "rename" with a warning given once to confirm that it doesnt copy over views, procedures & functions.

Is this a possibility?

**Comment by nathanpitman** [comment]
Yup, I'm seeing this too and it's a little frustrating. Would be great to see it fixed. :)

**Comment by dgm** [comment]
I have different data sets, and renaming databases is the easiest way to swap them in and out, but this is an annoying extra step

**Comment by anotherjames** [comment]
Looks like a duplicate of #1235?

**Comment by abhibeckert** [comment]
Not a duplicate, of 1235, just related.

**Comment by rowild** [comment]
This issue still is the case after 4 years.
